### PR TITLE
WIP: Refactoring+Removal of duplicate checks

### PR DESCRIPTION
**Pull request by hugbubby**
Changed the return value of del_groupchat to use
booleans. Removed an extraneous wipe_group_chat and replaced it with a
trim_groupchats function.

---
This change is [Reviewable](https://reviewable.io/reviews/toktok/c-toxcore/1193)

### CODE REVIEW

File: toxcore/tox.c [hunk 1]
```diff
@@ -1458,9 +1458,9 @@ uint32_t tox_conference_new(Tox *tox, Tox_Err_Conference_New *error)
 bool tox_conference_delete(Tox *tox, uint32_t conference_number, Tox_Err_Conference_Delete *error)
 {
     Messenger *m = tox->m;
-    int ret = del_groupchat(m->conferences_object, conference_number);
+    bool ret = del_groupchat(m->conferences_object, conference_number);
 
-    if (ret == -1) {
+    if (ret == true) {
```
Comment (iphydf) [inline toxcore/tox.c]: "true" means it failed?

File: toxcore/group.c [hunk 2]
```diff
@@ -141,33 +141,28 @@ static int32_t create_group_chat(Group_Chats *g_c)
     return id;
 }
 
-
-/* Wipe a groupchat.
- *
- * return -1 on failure.
- * return 0 on success.
+/*
```
Comment (iphydf) [inline toxcore/group.c]: I've been changing comments to doxygen style, so starting with `/**` and with `@return` and `@param`.

**Comment by hugbubby** [comment]
What would we do without you iphy

**Comment by iphydf** [comment]
@hugbubby how's this PR going?

**Comment by iphydf** [comment]
No updates here, no response from PR author. Closing for now. Feel free to reopen when you get to it.

**Pull request closed without merging**